src/sas/qtgui/Plotting/Slicers/ArcInteractor.py

```diff
@@ -0,0 +1,135 @@
+import numpy as np
+
+from sas.qtgui.Plotting.Slicers.BaseInteractor import BaseInteractor
+
+class ArcInteractor(BaseInteractor):
+    """
+    Draw an arc on a data2D plot with a variable radius (centered at [0,0]).
+    User interaction adjusts the parameter r
+
+    param r: radius from (0,0) of the arc on a data2D plot
+    param theta: angle from x-axis of the central point on the arc
+    param phi: angle from the centre point on the arc to each of its edges
+    """
+    def __init__(self, base, axes, color='black', zorder=5, r=1.0,
+                 theta=np.pi / 3, phi=np.pi / 8):
+        BaseInteractor.__init__(self, base, axes, color=color)
+        self.markers = []
+        self.axes = axes
+        self.color = color
+        # Variables for the current mouse position
+        self._mouse_x = r
+        self._mouse_y = 0
+        # Last known mouse position, for when the cursor moves off the plot
+        self._save_x = r
+        self._save_y = 0
+        self.scale = 10.0
+        # Key variables for drawing the interactor element
+        self.theta = theta
+        self.phi = phi
+        self.radius = r
+        # Calculate the marker coordinates and define the marker
+        marker_x = self.radius * np.cos(theta - 0.5 * phi)
+        marker_y = self.radius * np.sin(theta - 0.5 * phi)
+        self.marker = self.axes.plot([marker_x], [marker_y], linestyle='',
+                                     marker='s', markersize=10,
+                                     color=self.color, alpha=0.6, pickradius=5,
+                                     label='pick', zorder=zorder,
+                                     visible=True)[0]
+        # Define the arc
+        [self.arc] = self.axes.plot([], [], linestyle='-', marker='', color=self.color)
+        # The number of points that make the arc line
+        self.npts = 40
+        # Flag to keep track of motion
+        self.has_move = False
+        self.connect_markers([self.marker, self.arc])
+        self.update()
+
+    def set_layer(self, n):
+        """
+        Allow adding plot to the same panel
+        :param n: the number of layer
+        """
+        self.layernum = n
+        self.update()
+
+    def clear(self):
+        """
+        Clear this slicer and its markers
+        """
+        self.clear_markers()
+        try:
+            self.marker.remove()
+            self.arc.remove()
+        except:
+            # Old version of matplotlib
+            for item in range(len(self.axes.lines)):
```

Comment:
should we be worrying about this "old version" path anymore? `remove()` is available in >3.0, which we have standardized on for quite some time now.